src/Bonami/Collection/ArrayList.php

```diff
@@ -720,6 +722,25 @@ static function ($a, $b) use ($monoid) {
         );
     }
 
+    /**
+     * Converts items to numbers and then sums them up.
+     *
+     * Complexity: o(n)
+     *
+     * @see mfold - for folding diferent types of items (E.g. classes reprezenting BigNumbers and so on)
+     *
+     * @phpstan-param callable(T): (int|float) $itemToNumber
+     *
+     * @phpstan-return int|float
+     */
+    public function sum(callable $itemToNumber)
+    {
+        $sum = static function ($a, $b) {
+            return $a + $b;
+        };
+        return $this->map($itemToNumber)->reduce($sum, 0);
```

Comment:
Do we really want to do mapping to number first? Maybe it's root of all evil (ya know, premature optimisation), but we could do it just in one step (`reduce`).